CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/panes/QueryPhasePane.java

```diff
@@ -77,25 +81,31 @@ public class QueryPhasePane extends VBox {
      * @param top 
      * @param presetGlobalParms the current global plugin parameters
      */
-    public QueryPhasePane(final Map<String, List<DataAccessPlugin>> plugins,
-                          final PluginParametersPaneListener top,
+    public QueryPhasePane(final Map<String, Pair<Integer, List<DataAccessPlugin>>> plugins,                          final PluginParametersPaneListener top,
                           final PluginParameters presetGlobalParms) {
         globalParametersPane = new GlobalParametersPane(presetGlobalParms);
-
+        final List<Pair<Integer, HeadingPane>> orderedPlugins = new ArrayList<>();
         plugins.entrySet().stream()
-                .filter(pluginsOfType -> !pluginsOfType.getValue().isEmpty())
+                .filter(pluginsOfType -> !pluginsOfType.getValue().getValue().isEmpty())
                 .forEach(pluginsOfType -> {
                     final HeadingPane heading = new HeadingPane(
                             pluginsOfType.getKey(),
-                            pluginsOfType.getValue(),
+                            pluginsOfType.getValue().getValue(),
                             top,
                             globalParametersPane.getParamLabels()
                     );
-                    
-                    dataSourceList.getChildren().add(heading);
-                    dataSources.addAll(heading.getDataSources());
+
+            orderedPlugins.add(new Pair<>(pluginsOfType.getValue().getKey(), heading));
                 });
 
+        orderedPlugins.sort(Comparator.comparingInt(Pair<Integer, HeadingPane>::getKey));
+
+        for (int i = 0; i < orderedPlugins.size(); i++) {
```

Comment:
use a for each here instead. i.e. `for (final Pair<Integer, HeadingPane> plugin : orderedPlugins) {...`